provisioning/miniprov/hare_mp/cdf.py

```diff
@@ -24,17 +24,17 @@ def _get_dhall_path(self) -> str:
         raise RuntimeError('CFGEN Dhall types not found')
 
     def _gencdf(self) -> str:
-        resource_path = '/'.join(('dhall', 'gencdf.dhall'))
+        resource_path = 'dhall/gencdf.dhall'
         raw_content: bytes = pkg_resources.resource_string(
             'hare_mp', resource_path)
         return raw_content.decode('utf-8')
 
     def _create_node_descriptions(self) -> List[NodeDesc]:
         nodes: List[NodeDesc] = []
         conf = self.provider
-        server_dict: Dict[str, str] = conf.get('cluster>server_nodes')
-        for _, node in server_dict.items():
-            nodes.append(self._create_node(node))
+        machines: Dict[str, Any] = conf.get('server_node')
```

Comment:
sorry, don't see this key in the list of conf-store keys, didn't get how do we get the machines?

Reply:
The idea behind this is as follows.
The structure in the ConfStore is as follows `server_node>{machine-id}>everything-that-we-actually-need`. The problem is that there is no special key that lists all possible machine ids. While we need all of them to build CDF and specify all nodes.

So I tried this kind of hack, it turns out to be working with the current cortx-py-utils (installed from here http://cortx-storage.colo.seagate.com/releases/cortx/github/main/centos-7.8.2003/798/prod/cortx_iso/cortx-py-utils-2.0.0-182_6708fa0.noarch.rpm).

If there is a better way to work this around, I will be happy to follow.

Reply:
In other words, I just need to know all valid `machine_id`s to substitute into `server_node>{machine-id}>everything-that-we-actually-need`. So no, this is not a recursive fetching. It is just a way to find all possible machine-ids.

Reply:
>So I tried this kind of hack, it turns out to be working with the current cortx-py-utils (installed from here

Ah, so you mean for something like this,
```
{
  "server_node": {
    "SOME_MACHINE_ID": {
      "hostname": "myhost",
      "network": {
        "data": {
```
if you query just `server_node` it will give all the keys under `server_node` right? similar to `jq`. But if there are additional keys than `machine_id` under `server` then this will fail. Why not iterate over storage-sets and corresponding server_nodes, `cluster>{cluster_id}>storage_set[{storage_set_id}]>server_nodes` to get the machine_ids for the entire cluster? It may be a bit expensive but I think will be safer, no?

Reply:
Hm, yes, good idea. I 'll need to try it out on Monday.

Reply:
@mssawant But it means that we'll need to know `cluster_id` beforehand :-) 

Can we merge this PR (given that currently all the keys under `server_node` are machine IDs) now and probably create an issue for the future?

Perhaps ConfStore keys will change again and all this activity will not make much sense.

Reply:
Okay, if the current method is working fine and relevant tests are passing., then let's merge this and create a separate issue/ticket for improvmment.